pkg/util/provider/machinecontroller/machine_util.go

```diff
@@ -944,77 +944,94 @@ func (c *controller) setMachineTerminationStatus(ctx context.Context, deleteMach
 	return machineutils.ShortRetry, err
 }
 
-// getVMStatus tries to retrive VM status backed by machine
+// getVMStatus tries to retrieve VM status backed by machine
 func (c *controller) getVMStatus(ctx context.Context, getMachineStatusRequest *driver.GetMachineStatusRequest) (machineutils.RetryPeriod, error) {
```

Comment:
the function is named `getVMStatus` but it does not return any status. This is a bit weird. Shouldn't it be called `updateMachineStatusAndNodeLabel` instead?